package.json

```diff
@@ -72,7 +72,7 @@
     "parent-module": "^0.1.0",
     "path-to-regexp": "^0.1.2",
     "performance-now": "^2.1.0",
-    "protobufjs": "^6.9.0",
+    "pprof": "^3.0.0",
```

Comment:
We should not depend on this library since it may build which could be slow or even error.

Reply:
It generally won't since it uses node-gyp-prebuild. It's also required for the profiler to work.

Would making it an optional dependency be acceptable? I could do that and have the code bail with an error message if it's unable to load it. We _need_ to depend on this in _some_ way.

Reply:
There are no guarantees that the prebuilds will be available for every environment we support, and the module itself also doesn't support Node 8 so we simply cannot add a dependency on it right now. Having an optional dependency will still attempt to install/build the module for everyone. For now can we just document to do `npm install --save dd-trace pprof` and disable the profiler with an error log if `pprof` cannot be found?

Reply:
Making it optional will _attempt_ the install, but if it fails for any reason, such as being unsupported, the install will happily continue and ignore the failure. I don't see how it's an issue to add as an optional dependency.

Reply:
If it can fail, it means that the profiler may or may not work depending on the environment. Having to build also means it will slow down the installation significantly, which is not acceptable for many users, and could even cause crashes if it was able to build but built incorrectly. There are also issues with yarn in some cases where it will error out the entire install even when the issue was with an optional dependency. Adding this dependency would thus results in many potential problems that would not only affect profiling users but any user of dd-trace.

Reply:
Just tried installing the pprof module on my slow-ass 12" MacBook from 2015 and it takes 7 seconds to do a full build. I don't agree that it "will slow down the installation significantly".

As for crashing due to incorrect builds, has that actually ever happened? I've never heard of a gyp producing incorrect builds. 🤔 

Not really familiar with any yarn issues, do you have some more context you can share on that? I've not heard of that before either, but if it really is the case that an optional dependency could fail a yarn install, that's a bug in yarn.

Reply:
I've seen 7 seconds builds take minutes in some environments, so you can't assume based on your own machine which is anecdotal at best.

Crashing due to incorrect build has happened. At least 1 occurrence was because of a corrupted output, and there were many cases where something changed between the build and where the code actually ran, which is actually very common in Node.

Most of the issues with yarn are because of design choices that should not impact us in theory, but I've been bit in the past in some weird cases. Even if it's considered a bug, if a user hits the bug it still prevents them to install the tracer.

All the above issues were resolved for our own prebuilds of the runtime metrics native addons, but they apply to `pprof`.

Reply:
Marking as resolved as we'll defer dealing with this until just before going to public beta by putting this in a separate branch and making tagged prereleases.